### DISCUSSION TITLE
How can I migrate from a different local storage format to the one expected by zustand

### DISCUSSION THREAD

**Discussion by gregoryabailey** [pmndrs/zustand#2338] [General] [answered]
I am migrating from a localstorage value that just contains a string. The migrate function doesn't seem to fit as it expects to be migrating between versions that match the StateStorage interface. Should I use merge? hydration? something else?

**Comment by dai-shi** [comment, marked as answer]
Maybe a custom storage.

**Reply by gregoryabailey** [reply]
Thank you for the answer. I ended up just writing a migration script that reads the old value, removes it and sets the store default value to read value.
